Microsoft.Research/Clousot.Cache/SQLCacheModel.cs

```diff
@@ -558,6 +561,34 @@ public string CacheName
     {
       get { return this.cachename; }
     }
+
+    protected static void DetachDeletedDb(string connection)
+    {
+      // If anyone has manually deleted the cache file, it might be still registered with LocalDB. 
+      // If this is the case, we can't create a new database and the user is stuck unless he knows the tricky internals of LocalDB and how to get rid of the registration.
+      // => If we use LocalDB and the file does not exist, we try to detach it. 
+      try
+      {
+        var connectionString = new DbConnectionStringBuilder { ConnectionString = connection };
+        var fileName = (string)connectionString["AttachDbFileName"];
+        if (File.Exists(fileName)) 
+          return;
+
+        var catalog = (string)connectionString["Initial Catalog"];
+        var dataSource = (string)connectionString["Data Source"];
+
+        using (var master = new DataContext(string.Format(CultureInfo.InvariantCulture, @"Data Source={0};Initial Catalog=master;Integrated Security=True", dataSource)))
+        {
+          master.ExecuteCommand(@"exec sp_detach_db {0}", catalog);
+        }
+      }
+      catch
```

Comment:
:bulb: I prefer to see this list the actual exception types which are known to be thrown for the cases you describe.